src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +45,36 @@
      * Андрей - 600
      * София - 900</p>
      */
+    public static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+        LocalDate fromDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate toDate = LocalDate.parse(dateTo, formatter);
+
+        if (toDate.isBefore(fromDate)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        Map<String, Integer> salaries = new LinkedHashMap<>();
+        for (String name : names) {
+            salaries.put(name, 0);
+        }
```

Comment:
you are creating map here and putting only keys. And then you are replacing with these keys. Probably it's better to fill your map there?

Reply:
done!)